generic/tests/ping.py

```diff
@@ -58,76 +63,153 @@ def _get_loss_ratio(output):
+    if ping_ext_host:
+        logging.info("Ping test from guest side,"
+                           " dest: '%s'" % ext_host)
+        pkt_sizes = packet_sizes
+        # There is no ping program for guest, so let's hardcode...
+        cmd = ['ping']
+        cmd.append(ext_host)  # external host
+
+        is_windows = False
+        if params.get("os_type") == "windows":
+            cmd.append("-n 30")
+            cmd.append("-l %s")
+            # Windows doesn't support ping with packet larger than '65500'
+            pkt_sizes = [p for p in packet_sizes if int(p) < 65500]
+            # Add a packet size just equal '65500' for windows
+            pkt_sizes.append(65500)
+            is_windows = True
+        else:
+            cmd.append("-c 30")  # ping 30 times
+            cmd.append("-s %s")  # packet size
+            cmd.append("-i %s")  # time interval
+        cmd = " ".join(cmd)
+        for size in pkt_sizes:
+            if is_windows:
+                logging.info("Ping with packet size: %s" % size)
+                cur_cmd = cmd % size
+                status, output = session.cmd_status_output(cur_cmd, timeout=60)
                 _get_loss_ratio(output)
-
                 if status != 0:
                     raise error.TestFail(("Ping external host failed,"
                                           " status: %s, output: %s" %
                                           (status, output)))
+            else:
+                for interval_t in interval_times:
+                    logging.info("Ping with packet size: %s and interval: %s" % 
+                                 (size, interval_t))
+                    cur_cmd = cmd % (size, interval_t)
+                    status, output = session.cmd_status_output(cur_cmd, timeout=60)
+                    _get_loss_ratio(output)
+                    if status != 0:
+                        raise error.TestFail(("Ping external host failed,"
+                                              " status: %s, output: %s" %
+                                              (status, output)))
+
+        #ping with exceeded packages
+        logging.info("Ping test with exceeded package")
+        cmd = ['ping']
+        cmd.append(ext_host)
+
+        if is_windows:
+            cmd.append("-n 30")
+            cmd.append("-l 65508")
+            resultStr = "Bad value for option -l, valid range is from 0 to 65500"
+        else:
+            cmd.append("-c 30")
+            cmd.append("-s 65508")
+            resultStr = "packet size 65508 is too large. Maximum is 65507"
+        cmd = " ".join(cmd)
+        status, output = session.cmd_status_output(cmd, timeout=360)
+        if output.find(resultStr) == -1:
+            raise error.TestFail(("Ping test with exceeded package failed,"
+                                  " expected string: '%s' not found" % resultStr))
+       
+        #ping with flood
+        logging.info("Flood ping test from guest")
```

Comment:
It is suggested to use error_context.context("xxx", logging.info) for test steps prompts.